### DISCUSSION TITLE
What `defaultValues` default value can I use for an optional `Date` field?

### DISCUSSION THREAD

**Discussion by tibbe** [react-hook-form/react-hook-form#10825] [Q&A] [answered]
I have an optional `Date` field using a custom date picker component (in React Native). What's the correct default value to set for this field in `defaultValues`? Using `undefined` isn't recommended. Since `isDirty` depends on a comparison between the actual values and the default values I do need some sort of default value (see https://github.com/react-hook-form/react-hook-form/issues/3213).

**Comment by Moshyfawn** [comment]
Does `null` work for you?

**Comment by tibbe** [comment]
> Does `null` work for you?

Maybe. The question is really "what's commonly used?" or "what's best practice?". Optional fields seem like something that would be common but I have seen no docs around it, except those saying that default values can't be `undefined` (which would be a natural way to model optional fields in JavaScript).

**Reply by Moshyfawn** [reply]
`undefined` is sort of a reserved value in React. When it comes to the concept of controlled vs uncontrolled inputs in React, if you pass `undefined` to your input and later updated to a non-`undefined` value, you'll get a React warning "you can't mix controlled & uncontrolled inputs".

**Reply by Moshyfawn** [reply, marked as answer]
`null` is commonly used to supply an "empty" value to a number or a Date type of input.

**Reply by Moshyfawn** [reply]
For text inputs - an empty string is a good default; it follows the native HTML input conventions. Since RHF supports both controlled & uncontrolled inputs - following the Web standard of handling inputs is beneficial.

**Reply by tibbe** [reply]
Thanks for the explanation. To close the circle I think using `undefined` is more common in React Native, which I'm using, especially when representing missing values in custom components (as we're not tied to whatever HTML does). I will use `null`.
